src/dotty/tools/dotc/transform/PeepholeOptimize.scala

```diff
@@ -0,0 +1,25 @@
+package dotty.tools.dotc
+package transform
+
+import ast.{Trees, tpd}
+import core._, core.Decorators._
+import Contexts._, DenotTransformers._, TreeTransforms._, Phases.Phase
+import TreeExtractors._
+
+/** This phase performs safe peephole optimizations. */
+class PeepholeOptimize extends MiniPhaseTransform with IdentityDenotTransformer {
+  import tpd._
+
+  override def phaseName: String = "peepholeoptimize"
```

Comment:
I thought that maybe this could be extended with other peephole optimizations that are not specific to value classes, if you think that's a bad idea I could rename it to `ValueClassPeepholeOptimize`.